### DISCUSSION TITLE
Airflow DagProcessing Time Degradation

### DISCUSSION THREAD

**Discussion by oleg-ruban** [apache/airflow#30582] [Q&A]
After switching from 2.5.1 to 2.5.3 version I see a significant increase in the dag processing time (from less than one second to about 5 seconds per dag)
My assumptions were based on the metric [dag_processing.total_parse_time](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/metrics.html#:~:text=dag_processing.total_parse_time)
![dag_processing_total_parse_time](https://user-images.githubusercontent.com/54796035/231200846-fa90066f-526d-4160-b699-5abf74d4a3f6.png)
The peak on the graph corresponds to the use of version 2.5.3. Before and after - 2.5.1 version is used

The dag processor logs also show an increase in processing time for each dag separately:
For Airflow 2.5.1:
![dag_processing_251](https://user-images.githubusercontent.com/54796035/231201052-597aab9d-bca2-45fe-9f99-00a3ce752720.png)

For Airflow 2.5.3:
![dag_processing_253](https://user-images.githubusercontent.com/54796035/231201145-dc526703-74d0-42f2-b4cf-e3cdf182bd6c.png)

**Docker Image**
Custom docker image that is based on the official Airflow Python 3.8 Docker version 
- OS: debian
- Python 3.8

Additional APT Modules:
- locales
- tini
- gnupg
- tzdata
- openssl
- build-essential
- apt-transport-https
- ca-certificates
- acl
- libsasl2-dev
- libsasl2-2
- libsasl2-modules
- libsasl2-modules-gssapi-mit
- python3-dev
- libldap2-dev
- libssl-dev
- python3-ldap
- ldap-utils
- unixodbc-dev
- msodbcsql17
- libpq-dev
- openjdk-11-jre-headless
- git
- krb5-user
- krb5-config
- libgssapi-krb5-2
- libkrb5-dev
- sasl2-bin
- GCC

Additional PIP Modules:

- uests
- wheel
- python-json-logger
- setuptools
- statsd
- apache-airflow[statsd]
- apache-airflow-providers-hashicorp
- apache-airflow-providers-apache-spark
- apache-airflow-providers-microsoft-mssql
- apache-airflow-providers-telegram
- apache-airflow-providers-amazon
- apache-airflow-providers-microsoft-psrp
- apache-airflow-providers-microsoft-winrm
- apache-airflow-providers-redis
- apache-airflow-providers-oracle
- acryl-datahub-airflow-plugin
- pylint
- minio>=7.1.12
- pylint-airflow
- delta-spark
- psycopg2
- SQLAlchemy
- pyodbc
- kerberos
- requests-credssp

**Deployment environment**
The Airflow has been deployed to the Kubernetes Cluster (rke2 1.23.6) by using our own helm chart (much simplest version of the official Airflow Helm chart).
The DagFileProcessorManager runs as a standalone process by using env variable
```sh
- name: “AIRFLOW__SCHEDULER__STANDALONE_DAG_PROCESSOR”
  value: “True”
```
DagFileProcessorManager Pod has the following resources limits:
- cpu: "4"
- memory: 8Gi
- ephemeral-storage: 1Gi

Any thoughts on the reason for this?

**Comment by wookiist** [comment]
I'm going through the exact same situation. I've created an issue about it and we're talking about it.
https://github.com/apache/airflow/issues/30593
